### PR TITLE
Add Scheduled Push Template Notification Feature

### PR DESCRIPTION
**Pull request by skymorellici**
### What does this do and why?
Please include link to open issue if applicable.

### Additional notes for reviewers
* If applicable

### Testing
- [x] I wrote tests covering these changes  
- [x] I ran the full test suite and it passed

### Urban Airship Contribution Agreement
https://docs.urbanairship.com/contribution-agreement/

- [x] I've filled out and signed UA's contribution agreement form.

### Screenshots
* If applicable

### CODE REVIEW

**Comment by skymorellici** [comment]
Has this pull request planned to be reviewed, it's here from feb, anything missing?

**Comment by devinsmythe** [comment]
Hi @skymorellici we have a ticket created for tracking this issue and will look into it soon.

File: src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+package com.urbanairship.api.templates.model;
+
+
+import com.google.common.base.Preconditions;
+import com.urbanairship.api.push.model.DeviceTypeData;
+import com.urbanairship.api.push.model.audience.Selector;
+import com.urbanairship.api.schedule.model.Schedule;
+
+public class TemplateScheduledPushPayload {
+    private final Selector audience;
```
Comment (devinsmythe) [inline src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java]: An Optional name field can be set in the push payload so we should try and add support for that as well.
https://docs.airship.com/api/ua/#operation/api/templates/schedules/post/requestbody/items/name

File: src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void testTemplateScheduledPushPayloadSerializer() throws Exception {
+
+        TemplateSelector mergeData = TemplateSelector.newBuilder()
+                .setTemplateId("id123")
+                .addSubstitution("FIRST_NAME", "Prince")
+                .addSubstitution("LAST_NAME", "")
+                .build();
+
+        Schedule schedule = Schedule.newBuilder().setScheduledTimestamp(DateTime.now()).build();
```
[the remark this answers is not in the document]
Comment (skymorellici) [reply src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java]: done

File: src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+        TemplateScheduledPushPayload payload = TemplateScheduledPushPayload.newBuilder()
```
[the remark this answers is not in the document]
Comment (skymorellici) [reply src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java]: done

File: src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                "{" +
+                    "\"audience\":\"ALL\"," +
+                    "\"device_types\":[\"ios\",\"android\"]," +
+                    "\"merge_data\":{" +
+                        "\"template_id\":\"id123\"," +
+                        "\"substitutions\":{" +
+                            "\"FIRST_NAME\":\"Prince\"," +
+                            "\"LAST_NAME\":\"\"" +
+                        "}" +
+                    "}," +"\"schedule\":null"+
```
[the remark this answers is not in the document]
Comment (skymorellici) [reply src/test/java/com/urbanairship/api/templates/parse/TemplateScheduledPushPayloadSerializerTest.java]: done

File: src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+         * Build the TemplatePushPayload.
+         * <pre>
+         *     1. Audience, deviceTypes, and mergeData cannot be null.
+         * </pre>
+         *
+         * @return A TemplatePushPayload object
+         */
+        public TemplateScheduledPushPayload build() {
+            Preconditions.checkArgument(
+                    audience != null && deviceTypes != null && mergeData != null,
```
[the remark this answers is not in the document]
Comment (skymorellici) [reply src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java]: done

File: src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java (hunk 1 above)
Comment (skymorellici) [reply src/main/java/com/urbanairship/api/templates/model/TemplateScheduledPushPayload.java]: done

**Comment by skymorellici** [comment]
changes have been pushed, hope everything is ok.

**Comment by luminous-emanations** [comment]
Is this going to get merged anytime? Necessito por favor.

**Comment by pdxmele** [comment]
Hi, we're planning on merging in and making some small changes to do a release later this week.

**Comment by devinsmythe** [comment]
@luminous-emanations @skymorellici this was just released. Thanks for the contributions!

**Pull request merged**